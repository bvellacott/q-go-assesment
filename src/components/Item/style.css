.item_content {
	margin-right: 1rem;
}

.item_complete .item_content {
	color: #5f5;
}

.item_incomplete .item_content {
	color: #f85;
}