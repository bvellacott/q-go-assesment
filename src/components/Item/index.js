import React from 'react'
import './style.css'

const Item = ({ content, complete, remove, toggleComplete }) => (
	<span className={`item ${complete ? 'item_complete' : 'item_incomplete'}`}>
		<span className="item_content">{content}</span>
		{toggleComplete && (
		<button className="item_toggleCompleteButton" onClick={toggleComplete}>
			{complete ? 'set to doing' : 'set to done'}
		</button>)}
		{remove && (
		<button className="item_removeButton" onClick={remove}>X</button>)}
	</span>)

export default Item
