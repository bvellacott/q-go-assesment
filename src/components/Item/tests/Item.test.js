import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Item from '..'

describe('Item component', () => {
	const requiredProps = {}

	it('renders content span with content', () => {
		const wrapper = mount(<Item {...requiredProps} content="test todo"/>)

		const contentElement = wrapper.find('.item_content')
		expect(contentElement.length).toEqual(1)
		expect(contentElement.text()).toEqual('test todo')

		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('calls remove() when remove button clicked', () => {
		const remove = jest.fn()
		const wrapper = mount(<Item {...requiredProps} remove={remove}/>)

		const removeButton = wrapper.find('.item_removeButton')
		expect(removeButton.length).toEqual(1)

		removeButton.simulate('click')
		expect(remove).toHaveBeenCalled()

		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('calls toggleComplete() when toggle complete button clicked', () => {
		const toggleComplete = jest.fn()
		const wrapper = mount(<Item {...requiredProps} toggleComplete={toggleComplete}/>)

		const toggleCompleteButton = wrapper.find('.item_toggleCompleteButton')
		expect(toggleCompleteButton.length).toEqual(1)

		toggleCompleteButton.simulate('click')
		expect(toggleComplete).toHaveBeenCalled()

		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('does not render remove or toggleComplete button when handlers not passed', () => {
		const wrapper = mount(<Item {...requiredProps}/>)

		const removeButton = wrapper.find('.item_removeButton')
		expect(removeButton.length).toEqual(0)

		const toggleCompleteButton = wrapper.find('.item_toggleCompleteButton')
		expect(toggleCompleteButton.length).toEqual(0)

		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders complete class when complete', () => {
		const wrapper = mount(<Item {...requiredProps} complete={true}/>)
		const itemElement = wrapper.find('.item.item_complete')
		expect(itemElement.length).toEqual(1)

		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders incomplete class when incomplete', () => {
		const wrapper = mount(<Item {...requiredProps} complete={false}/>)
		const itemElement = wrapper.find('.item.item_incomplete')
		expect(itemElement.length).toEqual(1)

		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
