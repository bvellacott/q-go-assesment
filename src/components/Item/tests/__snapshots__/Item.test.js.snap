// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component calls remove() when remove button clicked 1`] = `
<Item
  remove={[Function]}
>
  <span
    className="item item_incomplete"
  >
    <span
      className="item_content"
    />
    <button
      className="item_removeButton"
      onClick={[Function]}
    >
      X
    </button>
  </span>
</Item>
`;

exports[`Item component calls toggleComplete() when toggle complete button clicked 1`] = `
<Item
  toggleComplete={[Function]}
>
  <span
    className="item item_incomplete"
  >
    <span
      className="item_content"
    />
    <button
      className="item_toggleCompleteButton"
      onClick={[Function]}
    >
      set to done
    </button>
  </span>
</Item>
`;

exports[`Item component does not render remove or toggleComplete button when handlers not passed 1`] = `
<Item>
  <span
    className="item item_incomplete"
  >
    <span
      className="item_content"
    />
  </span>
</Item>
`;

exports[`Item component renders complete class when complete 1`] = `
<Item
  complete={true}
>
  <span
    className="item item_complete"
  >
    <span
      className="item_content"
    />
  </span>
</Item>
`;

exports[`Item component renders content span with content 1`] = `
<Item
  content="test todo"
>
  <span
    className="item item_incomplete"
  >
    <span
      className="item_content"
    >
      test todo
    </span>
  </span>
</Item>
`;

exports[`Item component renders incomplete class when incomplete 1`] = `
<Item
  complete={false}
>
  <span
    className="item item_incomplete"
  >
    <span
      className="item_content"
    />
  </span>
</Item>
`;
